File: 349-intersection-of-two-arrays/intersection-of-two-arrays.py
# class Solution:
#     def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
#         res = []
#         nums1 = set(nums1)
#         for num in nums2:
#             if num in nums1:
#                 res.append(num)
#                 nums1.remove(num)
#         return res        

class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        nums1, nums2 = set(nums1), set(nums2)
        return list(nums1.intersection(nums2))

File: 349-intersection-of-two-arrays/intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        int m = nums1.size();
        int n = nums2.size();
        int i=0,j=0;
        vector<int> ans;

        sort(nums1.begin(),nums1.end());
        sort(nums2.begin(),nums2.end());
        
        while(i<m && j<n){
            if(nums1[i] == nums2[j]){
                int size = ans.size();
                if(size == 0){
                    ans.push_back(nums1[i]);
                }
                else if(ans[size-1] != nums1[i]){
                    ans.push_back(nums1[i]);
                } 
                i++;
                j++;
                
            }
            else if( nums1[i] > nums2[j]){
                j++;
            }
            else{
                i++;
            }
        }
        return ans;
    }
};